### PR TITLE
[CUMULUS-1964] Add provider param to Reports

### PR DESCRIPTION
**Pull request by bt6453**
**Summary:** Summary of changes

Addresses [CUMULUS-1964: Internal Reconciliation Report: Support list/array inputs for collectionId/Provider/GranuleId](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1964)

## Changes

* Reports can now take in a provider to filter on
* Can be array or just a single provider
* Turns report into a one-way report

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [X] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: packages/api/lib/reconciliationReport/normalizeEvent.js [hunk 1]
```diff
@@ -74,6 +74,33 @@ function updateCollectionIds(collectionId, reportType, modifiedEvent) {
...
+ * @param {Array<string>|string} provider - list of providers
+ * @param {string} reportType - report type
+ * @param {Object} modifiedEvent - input event
+ * @returns {Object} updated input even with correct provider and providers values.
+ */
+function updateProviders(provider, reportType, modifiedEvent) {
+  let returnEvent = { ...modifiedEvent };
+  if (provider) {
+    const providers = isString(provider) ? [provider] : provider;
+    if (reportType === 'Internal') {
```
Comment (jennyhliu) [inline packages/api/lib/reconciliationReport/normalizeEvent.js:89]: this is no longer needed.

File: packages/api/tests/lib/reconciliationReport/test-normalizeEvent.js [hunk 2]
```diff
@@ -152,14 +161,104 @@ test('normalizeEvent correctly handles granuleIds and collectionIds if reportTyp
...
   t.deepEqual(actual, expected);
 });
+
+test('normalizeEvent moves string on provider to array on providers', (t) => {
+  const inputEvent = {
+    systemBucket: 'systemBucket',
+    stackName: 'stackName',
+    startTimestamp: new Date().toISOString(),
+    endTimestamp: new Date().toISOString(),
+    reportType: 'Not Internal',
```
Comment (jennyhliu) [inline packages/api/tests/lib/reconciliationReport/test-normalizeEvent.js:184]: can be any type

File: packages/api/tests/lib/reconciliationReport/test-normalizeEvent.js [hunk 3]
```diff
@@ -152,14 +161,104 @@ test('normalizeEvent correctly handles granuleIds and collectionIds if reportTyp
...
+    ...inputEvent,
+    providers: ['provider1', 'provider2'],
+  };
+  delete expect.provider;
+
+  const actual = normalizeEvent(inputEvent);
+  t.deepEqual(actual, expect);
+});
+
+test('normalizeEvent throws error if array of providers is passed to Internal report', (t) => {
```
Comment (jennyhliu) [inline packages/api/tests/lib/reconciliationReport/test-normalizeEvent.js:217]: delete this test

**Review by jennyhliu — COMMENTED** [review]
As I merged https://github.com/nasa/cumulus/pull/1876, some special code for 'Internal' report is longer needed.

File: packages/api/lib/reconciliationReport/normalizeEvent.js (hunk 1 above)
Comment (bt6453) [reply packages/api/lib/reconciliationReport/normalizeEvent.js]: removed

File: packages/api/tests/lib/reconciliationReport/test-normalizeEvent.js (hunk 2 above)
Comment (bt6453) [reply packages/api/tests/lib/reconciliationReport/test-normalizeEvent.js:138]: I updated it to `anytype`

File: packages/api/tests/lib/reconciliationReport/test-normalizeEvent.js (hunk 3 above)
Comment (bt6453) [reply packages/api/tests/lib/reconciliationReport/test-normalizeEvent.js]: done!

File: packages/api/tests/lib/reconciliationReport/test-normalizeEvent.js [hunk 4]
```diff
@@ -110,14 +115,89 @@ test('normalizeEvent correctly handles granuleIds and collectionIds if reportTyp
...
+    ...inputEvent,
+    providers: ['provider1', 'provider2'],
+  };
+  delete expect.provider;
+
+  const actual = normalizeEvent(inputEvent);
+  t.deepEqual(actual, expect);
+});
+
+test('normalizeEvent throws error if providers and collectionIds are passed to non-Internal report', (t) => {
```
Comment (jennyhliu) [inline packages/api/tests/lib/reconciliationReport/test-normalizeEvent.js:171]: description doesn't match the test and error message

Comment (bt6453) [reply packages/api/tests/lib/reconciliationReport/test-normalizeEvent.js]: I updated it to use real examples and fix the description to use singular param-names.

**Review by jennyhliu — APPROVED** [review]
looks good!

**Comment by jennyhliu** [comment]
I forgot to mention, also need to update `cumulus-api` doc for `provider`, I was not sure about one-way comparison thing for provider, so let you update the docs.  Thanks.

**Pull request merged**